# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/MohammedTamer2025/Flyrank_ML_Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I will use Lane 2: Refresh / Content Opportunity Scoring.

The goal is to identify which content pages should be reviewed first for a possible refresh or improvement. I chose this lane because the dataset contains information about page performance, search visibility, engagement, freshness, and trends.

The goal is not simply to train a model. The goal is to use data to help prioritize which pages deserve human review.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

**Decision:** Which pages should be reviewed first for a possible content refresh or improvement?

**Action:** A content or SEO reviewer can investigate the highest-priority pages and decide whether to refresh, improve, protect, or monitor them.

**Cost of a wrong call:** A false positive could waste review time on a page that does not need attention. A false negative could cause an important page to be reviewed too late.

**Why data or ML can help:** Page performance depends on several signals. A data-driven scoring approach can combine these signals and help create a consistent priority list. ML will only be useful if it improves on a simple baseline.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")

import pandas as pd, numpy as np
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("\nTrend direction:")
print(df["trend_direction"].value_counts())

Number of rows: 30000

Trend direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


### What these numbers tell me

The dataset contains 30,000 content items. In the starter data, 16,262 items have a trend direction of `down`, while 5,962 are `stable` and 4,388 are `up`.

This shows that the dataset contains substantial variation in content trends, which makes it useful to investigate whether different signals can help prioritize pages for review.

## 4. Careful words: what I can and can't claim

I can identify patterns in the available data and use them to prioritize pages for human review.

I cannot claim that refreshing a particular page will definitely improve its traffic or rankings. The data is observational, so relationships in the data should not automatically be interpreted as causal effects.

I also cannot claim that the model predicts Google's ranking algorithm.

The eventual output should be treated as decision support: a ranked list of pages for human review, rather than an automatic decision.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.